src/ts/components/settingspanel.ts

```diff
@@ -367,17 +367,23 @@ export class SettingsPanel extends Container<SettingsPanelConfig> {
       if (item.isActive() && (item as any).setting instanceof SelectBox) {
         const selectBox = (item as any).setting as SelectBox;
         const oldDisplay = selectBox.getDomElement().css('display');
-        // updating the display to none marks the select-box as inactive, so it will be hidden with the rest
-        // we just have to make sure to reset this as soon as possible
-        selectBox.getDomElement().css('display', 'none');
-        if (window.requestAnimationFrame) {
-          requestAnimationFrame(() => {
+        if (oldDisplay !== 'none') {
```

Comment:
Optional: we're adding one more level here, could we simply early return, like this?

```ts
if (!item.isActive() || !((item as any).setting instanceof SelectBox)) {
  return;
}

const selectBox = (item as any).setting as SelectBox;
const oldDisplay = selectBox.getDomElement().css('display');

if (oldDisplay === 'none') {
  return;
}

selectBox.getDomElement().css('display', 'none');
// ...
```

Reply:
Agree, the suggested change will make code more readable. Updated code accordingly.